lib/handler/mruby/embedded/core.rb

```diff
@@ -25,6 +25,24 @@ def _h2o_eval_conf(__h2o_conf)
 end
 
 module Kernel
+  def task(&block)
+    f = Fiber.new do
+      block.call
+      # For when it's called in h2o_mruby_run_fiber and return output,
+      # or block doesn't have asynchronous callback
+      Fiber.yield([0, nil, nil])
+    end
+    fiber_res = f.resume()
+    # In case having no asynchronous callback function
+    if fiber_res[0] == 0
+      return
+    end
+    receiver = fiber_res[1]
+    klass = fiber_res[2][0]
+    # This should be called only one time.
+    # After that, the fiber is called in mruby_run_fiber and it register receiver in it.
+    klass.register_receiver(receiver, klass)
```

Comment:
>What does klass means? For example, in channel usage, fiber_res[2][0] seems to be H2O::Channel 's instance, not class.

I think we use `klass` for instance name in ruby usage. If others is better, I'll fix it.
https://stackoverflow.com/questions/4299289/what-is-the-difference-between-class-and-klass-in-ruby

Reply:
>Is it guaranteed that the klass object respond_to? :register_receiver ? I mean that many asynchronous functions can be called in task block, and HTTP response body streaming is one of the examples. So if I understand correctly, http_request(..).join[2].join fails (but sorry I haven't tried)

It's not guaranteed and we must create register_receiver when we create new class having async function. (I forgot to add for HttpInputStream, then I added here.)
https://github.com/h2o/h2o/pull/1336/commits/aa53e9023d33d26061c2674012b2a0c089e1964e

Reply:
>When the block yileds (calls defined callback method), there may be some args (i.e. fiber_res[2][1], fiber_res[2][2]..) How to handle that?

sorry, I realized `h2o_mruby_http_fetch_chunk_callback` are using args other than self in new code now.
How about this idea?:
1. move parts using args to new function named reserved word like `process_args(args)`
2. If there are some value on args other than self(index 0) in async function, call `process_args(args)`
3. do same thing for task function calling `process_args(args)`

For example, In case of `_h2o__http_fetch_chunk`,
1. move this(https://github.com/h2o/h2o/blob/master/lib/handler/mruby/http_request.c#L463-L471) from `h2o_mruby_http_fetch_chunk_callback` to new function `process_args`, and call `process_args`
2. In task function, if there is common same named function `process_args`.

In this way, we don't have to create branch in task function like mruby_run_fiber because of using same named function, but we have to care when we add function using args.